File: src/main/java/net/orandja/vw/mods/NerfedEntity/NerfedEntity.java
package net.orandja.vw.mods.NerfedEntity;

import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;

public interface NerfedEntity {

    short getTickCooldown();
    void setTickCooldown(short cooldown);

    boolean isMobDead();

    default void nerfAI(CallbackInfo info) {
        setTickCooldown((short) (getTickCooldown() - 1));
        if(getTickCooldown() > 0 && !isMobDead()) {
            info.cancel();
            return;
        }

        setTickCooldown((short) 30);
    }

}

File: src/main/kotlin/net/orandja/vw/logic/NerfedEntity.kt
package net.orandja.vw.logic

import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Inject
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo

interface NerfedEntity {

    var tickCooldown: Short
    var mobDead: Boolean

    @Inject(at = [At("HEAD")], method = ["tick"], cancellable = true)
    fun nerf(callbackInfo: CallbackInfo) {
        tickCooldown--
        if (tickCooldown > 0 && !this.mobDead) {
            callbackInfo.cancel()
            return
        }
        tickCooldown = 30
    }

}